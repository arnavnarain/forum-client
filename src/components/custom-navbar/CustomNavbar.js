import { Nav, Navbar, Container, NavDropdown } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Auth } from 'aws-amplify';

import "./custom-navbar.scss"
import { ComponentClassNames } from '@aws-amplify/ui-react';

const CustomNavbar = (props) => {
  const username = Auth.user.username;
  return (
    <Navbar bg="light" expand="md" sticky="top">
      <Container>
        <Navbar.Brand href="/home">Bulletin Board</Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="me-auto">
            <Nav.Link href="/home">Home</Nav.Link>
            <Nav.Link href="/about">About</Nav.Link>
            <Nav.Link href="/popular">Popular</Nav.Link>
          </Nav>
        </Navbar.Collapse>
        <Navbar.Collapse id="basic-navbar-nav" className={"dropdown"}>
          <NavDropdown title={username} id="basic-nav-dropdown">
            <NavDropdown.Item href="/myprofile">My Profile</NavDropdown.Item>
            <NavDropdown.Item onClick={props.signOut}>Sign out</NavDropdown.Item>
          </NavDropdown>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
}

export { CustomNavbar };

