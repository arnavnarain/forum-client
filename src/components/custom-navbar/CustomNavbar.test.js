import { render } from "@testing-library/react";
import { CustomNavbar } from './CustomNavbar';

jest.mock('@aws-amplify/ui-react', () => ({
    ComponentClassNames: {
      DROPDOWN: 'dropdown',
    },
  }));

  jest.mock('aws-amplify', () => ({
    Auth: {
      user: { username: 'mocked-username' },
    },
  }));
describe("(Snapshot) Default",  () => {
    it('(Snapshot) Default', async () => { 
        const { asFragment } = render(<CustomNavbar />)
        expect(asFragment()).toMatchSnapshot();
    })
})

