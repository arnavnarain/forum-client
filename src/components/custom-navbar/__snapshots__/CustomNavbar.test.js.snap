// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Default (Snapshot) Default 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-md navbar-light bg-light sticky-top"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/home"
      >
        Tennis Connect
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="me-auto navbar-nav"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="/home"
            href="/home"
          >
            Home
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="/live-ranking"
            href="/live-ranking"
          >
            Live Ranking
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="/live-matches"
            href="/live-matches"
          >
            Live Matches
          </a>
        </div>
      </div>
      <div
        class="dropdown navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="nav-item dropdown"
        >
          <a
            aria-expanded="false"
            class="dropdown-toggle nav-link"
            href="#"
            id="basic-nav-dropdown"
            role="button"
            tabindex="0"
          >
            mocked-username
          </a>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
